homepage/tests.py

```diff
@@ -103,3 +106,149 @@ def test_create_multiple_courses():
+@pytest.fixture
+def invalid_name():
+    return 'Pablo Diego Jose Francisco de Paula Juan Nepomuceno Maria de \
+    los Remedios Cipriano de la Santisima Trinidad Ruiz y Picasso Snape'
+
+
+# check the data stored in database from 0004_professor_test_data.py file
+@pytest.mark.django_db
+def test_model_professors_list(professors_list):
+
+    print(professors_list)
+    assert professors_list == [
+        ('Septima Vector',),
+        ('Sybill Patricia Trelawney',),
+        ('Bathsheda Babbling',),
+    ]
+
+
+@pytest.mark.django_db
+def test_add_valid_professor(valid_name):
+    professor = Professor(name=valid_name)
+    professor.save()
+
+    assert Professor.objects.filter(pk=professor.id).exists()
+
+
+# name is too long (>100)
+@pytest.mark.django_db
+def test_create_invalid_professor(invalid_name):
+    invalid = False
+    try:
+        Professor(invalid_name).clean_fields()
+    except ValidationError:
+        invalid = True
+
+    assert invalid
+
+
+@pytest.mark.django_db
+def test_professor_str(valid_name):
+    professor = Professor(name=valid_name)
+
+    print(str(professor))
+    print(professor.name)
+    assert str(professor) == professor.name
+
+
+# -----Professor to course tests-----#
+
+@pytest.fixture
+def professor_to_course_list():
+    return list(Professor_to_Course.objects.values_list(
+        'professor_id', 'course_id'))
+
+
+# check the data stored in database from 0004_professor_test_data.py file
+@pytest.mark.django_db
+def test_professor_to_course_list(professor_to_course_list):
+
+    print(professor_to_course_list)
+    assert professor_to_course_list == [
+        (1, 10221),
+        (1, 12357),
+        (2, 10231),
+        (3, 10111)
+    ]
+
+
+@pytest.mark.django_db
+def test_add_valid_pro_to_course():
+    relation = Professor_to_Course(
+        professor_id=Professor.objects.get(pk=2),
+        course_id=Course.objects.get(pk=10231))
+    relation.save()
+
+    assert Professor_to_Course.objects.filter(pk=relation.id).exists()
+
+
+# course does not exist (wrong id)
+@pytest.mark.django_db
+def test_add_invalid_course():
+    invalid = False
+
+    try:
+        Professor_to_Course(
+            Professor.objects.get(pk=2),
+            Course.objects.get(pk=10233)).clean_fields()
+    except ObjectDoesNotExist:
+        invalid = True
+
+    assert invalid
+
+
+# professor does not exist (wrong id)
+@pytest.mark.django_db
+def test_add_invalid_professor():
+    invalid = False
+
+    try:
+        Professor_to_Course(
+            Professor.objects.get(pk=5),
+            Course.objects.get(pk=10231)).clean_fields()
+    except ObjectDoesNotExist:
+        invalid = True
+
+    assert invalid
+
+
+@pytest.mark.django_db
+def test_get_pro_by_course():
+    professors_list = list(Professor_to_Course.get_professors_by_course(
+        Course.objects.get(pk=10231)))
+    professors = [arg[0] for arg in professors_list]  # retrieve pro ids
+
+    assert (len(professors) == 1) and (2 in professors)

```

Comment:
Maybe you can save ```2``` as a variable or add a comment to make it more clear it's professor's id